File: Daily Questions/2134-minimum-swaps-to-group-all-1s-together-ii/2134-minimum-swaps-to-group-all-1s-together-ii.c
int minSwaps(int* nums, int numsSize) {
    int nTotalOnes = 0;
    for(int i = 0; i < numsSize; i++) {
        nTotalOnes += nums[i];
    }
    if ((nTotalOnes == 0) || (nTotalOnes == numsSize)) {
        return 0;
    }
    int nOnesSW = 0;
    for(int i = numsSize - nTotalOnes; i < numsSize; i++) {
        nOnesSW += nums[i];
    }
    int nMaxOnesSW = nOnesSW;
    for(int i = 0; i < numsSize; i++) {
        if (numsSize-nTotalOnes+i < numsSize) {
            nOnesSW += nums[i] - nums[numsSize-nTotalOnes+i];
        }
        else {
            nOnesSW += nums[i] - nums[i-nTotalOnes];
        }
        if (nOnesSW > nMaxOnesSW) {
            nMaxOnesSW = nOnesSW;
        }
    }
    return (nTotalOnes - nMaxOnesSW);
}

File: B. Medium/2134-minimum-swaps-to-group-all-1s-together-ii/2134-minimum-swaps-to-group-all-1s-together-ii.py
class Solution:
    def minSwaps(self, nums: List[int]) -> int:
        n = len(nums)
        nTotalOnes = sum(nums)
        if (nTotalOnes == 0) or (nTotalOnes == n):
            return 0
        nMaxOnesSW = nOnesSW = sum(nums[-nTotalOnes:]) 
        for i in range(n):
            nOnesSW += nums[i] - nums[i-nTotalOnes]
            if nOnesSW > nMaxOnesSW:
                nMaxOnesSW = nOnesSW
        return nTotalOnes-nMaxOnesSW
        